server/api/unsafe_operation_test.go

```diff
@@ -83,3 +84,23 @@ func (suite *unsafeOperationTestSuite) TestRemoveFailedStores() {
 	err = tu.ReadGetJSON(re, testDialClient, suite.urlPrefix+"/remove-failed-stores/show", &output)
 	suite.NoError(err)
 }
+
+func (suite *unsafeOperationTestSuite) TestRemoveFailedStoresAutoDetect() {
+	re := suite.Require()
+
+	input := map[string]interface{}{"auto-detect": false}
+	data, _ := json.Marshal(input)
+	err := tu.CheckPostJSON(testDialClient, suite.urlPrefix+"/remove-failed-stores", data, tu.StatusNotOK(re),
+		tu.StringEqual(re, "\"Store ids are invalid\"\n"))
+	suite.NoError(err)
+
+	input = map[string]interface{}{"auto-detect": true}
+	data, _ = json.Marshal(input)
+	err = tu.CheckPostJSON(testDialClient, suite.urlPrefix+"/remove-failed-stores", data, tu.StatusOK(re))
+	suite.NoError(err)
+
+	// Test show
+	var output []cluster.StageOutput
+	err = tu.ReadGetJSON(re, testDialClient, suite.urlPrefix+"/remove-failed-stores/show", &output)
+	suite.NoError(err)
```

Comment:
Suggest checking the output, not just the error.

Reply:
checking the output doesn't make any sense, the stage doesn't reach `finished` here